source/extensions/filters/http/common/jwks_fetcher.cc

```diff
@@ -66,38 +105,42 @@ class JwksFetcherImpl : public JwksFetcher,
     ENVOY_LOG(trace, "{}", __func__);
     complete_ = true;
     const uint64_t status_code = Http::Utility::getResponseStatus(response->headers());
+
+    JwksFetcher::JwksReceiver::Failure reason = JwksFetcher::JwksReceiver::Failure::Network;
+
     if (status_code == enumToInt(Http::Code::OK)) {
-      ENVOY_LOG(debug, "{}: fetch pubkey [uri = {}]: success", __func__, uri_->uri());
+      ENVOY_LOG(debug, "{}: fetch pubkey [uri = {}]: success", __func__, uri_.uri());
       if (response->body().length() != 0) {
         const auto body = response->bodyAsString();
         auto jwks =
             google::jwt_verify::Jwks::createFrom(body, google::jwt_verify::Jwks::Type::JWKS);
         if (jwks->getStatus() == google::jwt_verify::Status::Ok) {
-          ENVOY_LOG(debug, "{}: fetch pubkey [uri = {}]: succeeded", __func__, uri_->uri());
+          ENVOY_LOG(debug, "{}: fetch pubkey [uri = {}]: succeeded", __func__, uri_.uri());
           receiver_->onJwksSuccess(std::move(jwks));
+          reset();
+          return;
         } else {
-          ENVOY_LOG(debug, "{}: fetch pubkey [uri = {}]: invalid jwks", __func__, uri_->uri());
-          receiver_->onJwksError(JwksFetcher::JwksReceiver::Failure::InvalidJwks);
+          ENVOY_LOG(debug, "{}: fetch pubkey [uri = {}]: invalid jwks", __func__, uri_.uri());
+          reason = JwksFetcher::JwksReceiver::Failure::InvalidJwks;
```

Comment:
We should not retry invalid Jwks.  It will not help

Reply:
done